src/TwigComponent/src/Resources/doc/index.rst

```diff
@@ -637,14 +637,81 @@ the twig template and twig variables before components are rendered::
         }
     }
 
-Embedded Components
--------------------
+Nested Components
+-----------------
 
-It's totally possible to embed one component into another. When you do
+It's totally possible to nest one component into another. When you do
 this, there's nothing special to know: both components render
 independently. If you're using `Live Components`_, then there
 *are* some guidelines related to how the re-rendering of parent and
-child components works. Read `Live Embedded Components`_.
+child components works. Read `Live Nested Components`_.
+
+Embedded Components
+-------------------
+
+.. versionadded:: 2.2
+
+    Embedded components were added in TwigComponents 2.2.
+
+You can write your component's Twig template with blocks that can be overridden
+when rendering using the ``{% component %}``. These blocks can be thought of as
+*slots* which you may be familiar with from Vue. The ``component`` tag is very
+similar to Twig's native `embed tag`_.
+
+Consider a data table component. You pass it headers and rows but can expose
+blocks for the cells and an optional footer:
+
+.. code-block:: twig
+
+    {# templates/components/data_table.html.twig #}
+
+    <div{{ attributes.defaults({class: 'data-table'}) }}>
+        <table>
+            <thead>
+                <tr>
+                    {% for header in this.headers %}
+                        <th>
+                            {% block th %}{{ header }}{% endblock %}
+                        </th>
+                    {% endfor %}
+                </tr>
+            </thead>
+            <tbody>
+                {% for row in this.data %}
+                    <tr>
+                        {% for cell in row %}
+                            <td>
+                                {% block td %}{{ cell }}{% endblock %}
+                            </td>
+                        {% endfor %}
+                    </tr>
+                {% endfor %}
+            </tbody>
+        </table>
+        {% if block('footer') is defined %}
```

Comment:
I think we should make this a tiny bit less fancy. Just:

```twig
{% block footer %}{% endblock %}
```

Then below, when we render the component, we include the entire `<div class="data-table-footer">`.

My think is that, the main & simplest use case (I think) for this is where you have empty blocks that you want to fill in (e.g. filling in the content of an `alert` component). And currently, we only show more advanced use-cases (calling `parent()` & using a block with `is defined`).